### PR TITLE
MONGOCRYPT-399 Process payload `FLE2EncryptionPlaceholder` to `FLE2UnindexedEncryptedValue`

### PR DESCRIPTION
**Pull request by kevinAlbs**

### CODE REVIEW

File: src/mc-fle2-payload-uev.c [hunk 1]
```diff
@@ -172,6 +172,52 @@ mc_FLE2UnindexedEncryptedValue_decrypt (_mongocrypt_crypto_t *crypto,
...
+                                        const _mongocrypt_buffer_t *key,
+                                        _mongocrypt_buffer_t *out,
+                                        mongocrypt_status_t *status)
+{
+   _mongocrypt_buffer_t iv = {0};
+   bool res = false;
+
+   _mongocrypt_buffer_resize (&iv, MONGOCRYPT_IV_LEN);
+   if (!_mongocrypt_random (crypto, &iv, MONGOCRYPT_IV_LEN, status)) {
+      goto fail;
```
Comment (vector-of-bool) [inline src/mc-fle2-payload-uev.c:189]: This `goto` will attempt to cleanup the `AD` variable before control has passed over its zero-initialization below.

File: src/mongocrypt-marking.c [hunk 2]
```diff
@@ -611,6 +612,53 @@ _mongocrypt_fle2_placeholder_to_find_ciphertext (
...
+{
+   _mongocrypt_buffer_t plaintext = {0};
+   mc_FLE2EncryptionPlaceholder_t *placeholder = &marking->fle2;
+   _mongocrypt_buffer_t user_key = {0};
+   bool res = false;
+
+   BSON_ASSERT (marking->type == MONGOCRYPT_MARKING_FLE2_ENCRYPTION);
+   BSON_ASSERT (placeholder->algorithm == MONGOCRYPT_FLE2_ALGORITHM_UNINDEXED);
+   _mongocrypt_ciphertext_init (ciphertext);
+   _mongocrypt_buffer_init (&plaintext);
```
Comment (vector-of-bool) [inline src/mongocrypt-marking.c:630]: I think this `buffer_init` is redundant with the immediate `buffer_from_iter` on the next line.

**Review by vector-of-bool — COMMENTED** [review]
Looks good, but two items feel suspicious.

File: src/mc-fle2-payload-uev.c (hunk 1 above)
Comment (kevinAlbs) [reply src/mc-fle2-payload-uev.c:190]: Good catch! Moved the declaration of `AD` to the start.

File: src/mongocrypt-marking.c (hunk 2 above)
Comment (kevinAlbs) [reply src/mongocrypt-marking.c]: Removed.

**Review by vector-of-bool — APPROVED** [review]
LGTM

**Pull request merged**